### PR TITLE
feat: [CI-9117]: Ruby runner sync with latest agent

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53190)

### CODE REVIEW

File: product/ci/addon/tasks/runtests.go [hunk 1]
```diff
@@ -581,20 +582,20 @@ func (r *runTestsTask) getCmd(ctx context.Context, agentPath, outputVarFile stri
 			return "", err
 		}
 		agentArg = fmt.Sprintf(javaAgentArg, iniFilePath)
+		tiPreCmd += fmt.Sprintf("export TMPDIR=%s\nexport HARNESS_JAVA_AGENT=%s\n", r.tmpFilePath, agentArg)
```
Comment (rutvijmehta-harness) [inline product/ci/addon/tasks/runtests.go:585]: `TMPDIR` is not required for python and csharp?

File: product/ci/addon/tasks/runtests.go [hunk 2]
```diff
@@ -573,6 +573,7 @@ func (r *runTestsTask) getCmd(ctx context.Context, agentPath, outputVarFile stri
 
 	// Config file
 	var iniFilePath, agentArg string
+	tiPreCmd := "set -xe\n"
```
Comment (rutvijmehta-harness) [inline product/ci/addon/tasks/runtests.go:576]: IMO we should move this logic to each runner. No required in this PR but we should have a ticket.

File: product/ci/addon/testintelligence/ruby/rspec.go [hunk 3]
```diff
@@ -64,11 +64,13 @@ func (b *rspecRunner) ReadPackages(files []types.File) []types.File {
 
 func (b *rspecRunner) GetCmd(ctx context.Context, tests []types.RunnableTest, userArgs, agentConfigPath string, ignoreInstr, runAll bool) (string, error) {
 	testCmd := ""
-	tiFlag := "INTEL=1"
+	tiFlag := "TI=1 TI_OUTPUT_PATH=/usr/src/myapp/cg/"
```
Comment (rutvijmehta-harness) [inline product/ci/addon/testintelligence/ruby/rspec.go:67]: Should we not export this in the precommand above?

File: product/ci/addon/tasks/runtests.go (hunk 1 above)
Comment (jamie-harness) [reply product/ci/addon/tasks/runtests.go:585]: I think it's only for bazel

File: product/ci/addon/testintelligence/ruby/rspec.go (hunk 3 above)
Comment (jamie-harness) [reply product/ci/addon/testintelligence/ruby/rspec.go:67]: oh sorry this is a debug statement. I will remove the TI_OUTPUT_PATH=/usr/src/myapp/cg/ part

Comment (jamie-harness) [reply product/ci/addon/testintelligence/ruby/rspec.go:67]: but TI part is needed to enable/disable agent

File: product/ci/addon/tasks/runtests.go (hunk 2 above)
Comment (jamie-harness) [reply product/ci/addon/tasks/runtests.go:576]: It should be more like each language. We should probably have a language level abstraction

Comment (rutvijmehta-harness) [reply product/ci/addon/tasks/runtests.go:576]: Okay, let's create a ticket anyway. The switch cases will keep growing.

File: product/ci/addon/testintelligence/ruby/rspec.go (hunk 3 above)
Comment (rutvijmehta-harness) [reply product/ci/addon/testintelligence/ruby/rspec.go:67]: Sounds good.

**Review by rutvijmehta-harness — APPROVED** [review]

**Pull request merged**